django/models/README.md

```diff
@@ -0,0 +1,263 @@
+# Model
+
+Best practices for creating model class.
+
+## Table of Contents
+
+- [Common](#common)
+- [Fields](#fields)
+- [Model Controller](#model-controller)
+- [Managers](#managers)
+
+## Common
+
+These are some of the best practices that you should add to your model class.
+
+### The String Representation Function
+
+By default django will use id when we try to print the object.
+
+```python
+
+obj = Model.objects.first()
+print(obj)
+>>> <Model: (1)>
+```
+
+It's easier to debug when you get more information regarding the object.
+
+```python
+class Person(models.Model):
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    position = models.ForeignKey(
+        to='people.Position',
+        on_delete=models.CASCADE,
+        related_name='people',
+    )
+
+    # String function that you should add
+    def __str__(self):
+        return f'{self.first_name} {self.last_name}'
+```
+
+When trying to debug:
+
+```python
+person = Person.objects.first()
+print(person)
+>>> <Person: Khemanorak Khath>
+```
+
+**CAUTION**: You should not use foreign key field in your string function,
+because this could lead to performance problem.
+
+```python
+# No
+def __str__(self):
+    return f'{self.first_name} {self.last_name} {self.position.name}'
+
+# If your query is not select_related position, this will hit database
+# with the amount of people.
+```
+
+## Fields
+
+Best practices when defining field in model.
+
+### Naming
+
+- Field name should be **noun** and in **singular** form, except
+for many-to-many field should be **plural** form.
+- Boolean field should be **verb** with past tense.
+- Do not repeat model name in field name.
+- Related name parameter value should be **plural** form.
+
+```python
+class Person(models.Model):
+    # Yes: Singular noun
+    first_name = models.CharField(max_length=100)
+    # No : Repeat model name person
+    person_first_name = models.CharField(max_length=100)
+    
+    # Yes: Many to Many in plural form
+    companies = model.ManyToMany(to='company.Company', blank=True, related_name='people')
+
+    # Yes
+    activated = models.BooleanField()
+    # No
+    is_active = models.BooleanFiled()
+```
+
+### Relationship
+
+- When defining foreign key, one to one or many to many in a model it should be
+lazy reference instead of importing the model class directly, preventing
+circular or recursive import exception.
+[Read more](https://docs.djangoproject.com/en/3.1/ref/models/fields/#foreignkey)
+- Always add related name parameter to foreign key or many to many field. If you
+don't want Django to create backward relation add '+' to the `related_name` parameter.
```

Comment:
put comma in the sentences

If you don't want Django to create backward relation, add '+' to the `related_name` parameter.